requirements-dev.txt

```diff
@@ -17,20 +17,19 @@ lxml==4.9.2
 
 # This version is mentioned in `site/source/docs/site/about.rst`.
 # Please keep them in sync.
-sphinx==2.4.4
+sphinx==7.1.2
 # See https://github.com/emscripten-core/emscripten/issues/21590
-sphinxcontrib-applehelp<=1.0.4
-sphinxcontrib-devhelp<=1.0.2
-sphinxcontrib-htmlhelp<=2.0.0
-sphinxcontrib-serializinghtml<=1.1.5
-sphinxcontrib-qthelp<=1.0.3
-alabaster<=0.7.12
+sphinxcontrib-applehelp==1.0.4
+sphinxcontrib-devhelp==1.0.2
+sphinxcontrib-htmlhelp==2.0.1
+sphinxcontrib-serializinghtml==1.1.5
+sphinxcontrib-qthelp==1.0.3
+sphinxcontrib-jquery==4.0
+alabaster==0.7.13
 pygments==2.17.2
-# See https://github.com/readthedocs/readthedocs.org/issues/9038
-jinja2<3.1
+jinja2==3.0
 
-# Pin docutils because newer versions are not compatible with sphinx 2.4.4
-docutils==0.17.1
```

Comment:
How about if we simply remove all of these dependencies except for the sphinx one ones?

Reply:
I saw that `sphinx 7.1.2` has `jinja >= 3.0` as a requirement.

I think it's best to keep `jinja2==3.0` to avoid future releases of `jinja2` to break things.

What do you think?

Reply:
I think its best to only include direct dependencies here, up until the point where we have some reason to do otherwise.

But i'm no expert so if you think this is safer/better then I'm fine with it.

How about the rest of these indirect deps, can  we remove those?